aspnet/publishing/linuxproduction.rst

```diff
@@ -0,0 +1,190 @@
+
+A reverse proxy is a common setup for serving dynamic web applications. The reverse proxy terminates the HTTP request and forwards it to the ASP.NET application.
+
+Why use a reverse-proxy server?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Kestrel is great for serving dynamic content from ASP.NET, however the web serving parts aren’t as feature rich full-featured servers like IIS, Apache or Nginx. A reverse proxy-server can allow you to offload work like serving static content, caching requests, compressing requests, and SSL termination from the HTTP server. The reverse proxy server may reside on a dedicated machine or may be deployed alongside an HTTP server.
+
+For the purposes of this guide, we are going to use a single instance of Nginx that runs on the same server alongside your HTTP server. However, based on your requirements you may chose a different setup.
+
+Install Nginx
+~~~~~~~~~~~~~
+
+.. code-block:: bash
+
+    sudo apt-get install nginx
+
+.. note::
+
+    If you plan to install optional Nginx modules you may be required to
+    build Nginx from source.
+
+We are going to ``apt-get`` to install Nginx. The installer also creates a System V init script that runs Nginx as daemon on system startup. Since we just installed Nginx for the first time, we can explicitly start it by running
+
+.. code-block:: bash
+
+    sudo service nginx start
+
+At this point you should be able to navigate to your browser and see the default landing page for Nginx.
+
+Configure Nginx
+~~~~~~~~~~~~~~~
+
+We will now configure Nginx as a reverse proxy to forward requests to our ASP.NET application
+
+We will be modifying the ``/etc/nginx/sites-available/default``, so open it up in your favorite text editor and replace the contents with the following.
+
+.. code-block:: nginx
+   :emphasize-lines: 7
+
+    server {
+        listen 80;
+        location / {
+            proxy_pass http://unix:/var/aspnet/HelloMVC/kestrel.sock;
+            proxy_http_version 1.1;
+            proxy_set_header Upgrade $http_upgrade;
+            proxy_set_header Connection keep-alive;
+            proxy_set_header Host $host;
+            proxy_cache_bypass $http_upgrade;
+        }
+    }
+
+This is one of the simplest configuration files for Nginx that forwards incoming public traffic on your port ``80`` to a unix socket that your web application will listen on. You can specify this Unix socket in your ``project.json`` file.
+
+.. code-block:: json
+    :caption: project.json (truncated)
+
+    "commands": {
+        "web": "Microsoft.AspNet.Server.Kestrel --server.urls http://unix:/var/aspnet/HelloMVC/kestrel.sock",
+    },
+
+.. note::
+
+    The ``proxy_set_header Connection keep-alive;`` is **required** as a temporary workaround a `known bug in Kestrel <https://github.com/aspnet/KestrelHttpServer/issues/341>`_.
+
+You might want to look at ``etc/nginx/nginx.conf`` to make other necessary changes to your nginx environment to configure your nginx environment.
```

Comment:
Nit: `etc/nginx/nginx.conf` should be `/etc/nginx/nginx.conf` (first `/` missing). Also "nginx" should have a capital N.